orchestration/dags/dependencies/Offer_name_tags.py

```diff
@@ -0,0 +1,122 @@
+import gcsfs
+import pandas as pd
+import nltk
+import re
+import json
+from nltk.corpus import stopwords as StopWords
+import collections
+from dependencies.bigquery_client import BigQueryClient
+from dependencies.config import (
+    GCP_PROJECT,
+    BIGQUERY_CLEAN_DATASET,
+    BIGQUERY_ANALYTICS_DATASET,
+    DATA_GCS_BUCKET_NAME,
+)
+
+nltk.download("stopwords")
+stopwords = StopWords.words("french")
+
+
+def get_offers_name_to_tags():
+    query = f"""SELECT offer_name, offer_id 
+                FROM {GCP_PROJECT}.{BIGQUERY_CLEAN_DATASET}.applicative_database_offer """
+    offer_name_to_tag = pd.read_gbq(query)
+    return offer_name_to_tag
+
+
+def clean(name):
+    name = re.sub(r"[^\w\s]", "", name.lower())
+    return " ".join([w for w in name.split() if w not in stopwords])
+
+
+def tokenize(string):
+    """Convert string to lowercase and split into words (ignoring
+    punctuation), returning list of words.
+    """
+    return re.findall(r"\w+", string.lower())
+
+
+def map_common_ngrams(df, most_common, min_length=2, max_length=4):
+    """Iterate through given lines iterator (file object or list of
+    lines) and check if the line contains a common n-gram, if so, the
+    n-gram (tag) is added in the associated column of the dataframe.
+    """
+    lines = df["offer_name_clean_stop"]
+    lengths = range(min_length, max_length + 1)
+    queue = collections.deque(maxlen=max_length)
+
+    # Helper function to add n-grams at start of current queue to dict
+    def add_queue():
+        current = tuple(queue)
+        for length in lengths:
+            if len(current) >= length:
+                # If the n-gram is contained in the list of common n-grams
+                if current[:length] in most_common[len(current[:length]) - 2]:
```

Comment:
Quand je teste le DAG j'ai l'erreur "IndexError: list index out of range" à cette ligne